plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSortingFileWriter.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import io.airlift.units.DataSize;
+import io.trino.filesystem.TrinoFileSystem;
+import io.trino.plugin.hive.SortingFileWriter;
+import io.trino.spi.Page;
+import io.trino.spi.PageSorter;
+import io.trino.spi.connector.SortOrder;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.TypeOperators;
+import org.apache.hadoop.fs.Path;
+import org.apache.iceberg.Metrics;
+
+import java.io.Closeable;
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class IcebergSortingFileWriter
+        implements IcebergFileWriter
+{
+    private final IcebergFileWriter outputWriter;
+    private final SortingFileWriter sortingFileWriter;
+
+    public IcebergSortingFileWriter(
+            TrinoFileSystem fileSystem,
+            Path tempFilePrefix,
+            IcebergFileWriter outputWriter,
+            DataSize maxMemory,
+            int maxOpenTempFiles,
+            List<Type> types,
+            List<Integer> sortFields,
+            List<SortOrder> sortOrders,
+            PageSorter pageSorter,
+            TypeOperators typeOperators,
+            SortingFileWriter.TempFileSinkFactory tempFileSinkFactory)
```

Comment:
`SortingFileWriter.TempFileSinkFactory tempFileSinkFactory` this is an implementation detail.
Initialize directly the `tempFileSinkFactory` while creating the `SortingFileWriter` and remove this constructor parameter.